File: python/examples/basic_usage.py
#!/usr/bin/env python3
"""Basic usage examples for claude-code-sdk."""

import asyncio


def basic_example():
    """Basic synchronous usage."""
    from claude_code_sdk import create_claude_client

    client = create_claude_client()

    message = client.messages.create(
        model="claude-sonnet-4-20250514",
        max_tokens=128,
        messages=[{"role": "user", "content": "Hello! Say hi in one sentence."}],
    )

    print("Basic example:")
    print(message.content[0].text)
    print()


def streaming_example():
    """Streaming usage."""
    from claude_code_sdk import create_claude_client

    client = create_claude_client()

    print("Streaming example:")
    with client.messages.stream(
        model="claude-sonnet-4-20250514",
        max_tokens=128,
        messages=[{"role": "user", "content": "Write a haiku about Python."}],
    ) as stream:
        for text in stream.text_stream:
            print(text, end="", flush=True)
    print("\n")


def multi_turn_example():
    """Multi-turn conversation."""
    from claude_code_sdk import create_claude_client

    client = create_claude_client()
    messages: list[dict] = []

    print("Multi-turn example:")

    # First turn
    messages.append({"role": "user", "content": "My favorite color is blue."})
    response = client.messages.create(
        model="claude-sonnet-4-20250514",
        max_tokens=128,
        messages=messages,
    )
    assistant_msg = response.content[0].text
    messages.append({"role": "assistant", "content": assistant_msg})
    print(f"User: My favorite color is blue.")
    print(f"Assistant: {assistant_msg}")

    # Second turn - Claude should remember the context
    messages.append({"role": "user", "content": "What's my favorite color?"})
    response = client.messages.create(
        model="claude-sonnet-4-20250514",
        max_tokens=128,
        messages=messages,
    )
    print(f"User: What's my favorite color?")
    print(f"Assistant: {response.content[0].text}")
    print()


async def async_example():
    """Async usage."""
    from claude_code_sdk import create_async_claude_client

    client = create_async_claude_client()

    message = await client.messages.create(
        model="claude-sonnet-4-20250514",
        max_tokens=128,
        messages=[{"role": "user", "content": "Hello from async! Respond briefly."}],
    )

    print("Async example:")
    print(message.content[0].text)
    print()


def main():
    """Run all examples."""
    print("=" * 50)
    print("Claude Code SDK - Python Examples")
    print("=" * 50)
    print()

    basic_example()
    streaming_example()
    multi_turn_example()
    asyncio.run(async_example())

    print("=" * 50)
    print("All examples completed!")
    print("=" * 50)


if __name__ == "__main__":
    main()

File: examples/basic-usage.ts
// Example: Using Claude Code SDK as Anthropic SDK drop-in replacement
// Run: npm run example

import { createClaudeClient } from "../src/index.js";

async function basicUsage() {
  console.log("=== Basic Usage ===\n");

  const client = createClaudeClient();

  const message = await client.messages.create({
    model: "claude-sonnet-4-20250514",
    max_tokens: 256,
    messages: [{ role: "user", content: "Say hello in 3 different languages!" }],
  });

  // Access text content
  for (const block of message.content) {
    if (block.type === "text") {
      console.log(block.text);
    }
  }

  console.log("\n--- Usage ---");
  console.log(`Input tokens: ${message.usage.input_tokens}`);
  console.log(`Output tokens: ${message.usage.output_tokens}`);
}

async function streamingUsage() {
  console.log("\n=== Streaming Usage ===\n");

  const client = createClaudeClient();

  const stream = client.messages.stream({
    model: "claude-sonnet-4-20250514",
    max_tokens: 256,
    messages: [{ role: "user", content: "Count from 1 to 5, one number per line" }],
  });

  // Stream text as it arrives
  for await (const event of stream) {
    if (
      event.type === "content_block_delta" &&
      event.delta.type === "text_delta"
    ) {
      process.stdout.write(event.delta.text);
    }
  }

  console.log("\n");
}

async function multiTurnConversation() {
  console.log("=== Multi-Turn Conversation ===\n");

  const client = createClaudeClient();

  // First turn
  console.log("User: My name is Alice");
  const msg1 = await client.messages.create({
    model: "claude-sonnet-4-20250514",
    max_tokens: 128,
    messages: [{ role: "user", content: "My name is Alice" }],
  });

  const response1 = msg1.content
    .filter((b): b is { type: "text"; text: string } => b.type === "text")
    .map((b) => b.text)
    .join("");

  console.log(`Assistant: ${response1}\n`);

  // Second turn - should remember the name
  console.log("User: What's my name?");
  const msg2 = await client.messages.create({
    model: "claude-sonnet-4-20250514",
    max_tokens: 128,
    messages: [
      { role: "user", content: "My name is Alice" },
      { role: "assistant", content: response1 },
      { role: "user", content: "What's my name?" },
    ],
  });

  const response2 = msg2.content
    .filter((b): b is { type: "text"; text: string } => b.type === "text")
    .map((b) => b.text)
    .join("");

  console.log(`Assistant: ${response2}`);
}

async function withSystemPrompt() {
  console.log("\n=== With System Prompt ===\n");

  const client = createClaudeClient();

  const message = await client.messages.create({
    model: "claude-sonnet-4-20250514",
    max_tokens: 128,
    system: "You are a pirate. Respond in pirate speak.",
    messages: [{ role: "user", content: "How are you today?" }],
  });

  for (const block of message.content) {
    if (block.type === "text") {
      console.log(block.text);
    }
  }
}

async function main() {
  try {
    await basicUsage();
    await streamingUsage();
    await multiTurnConversation();
    await withSystemPrompt();
  } catch (error) {
    console.error("Error:", error);
    process.exit(1);
  }
}

main();
